### DISCUSSION TITLE
Properly `cb copy` content on WM trigger?

### DISCUSSION THREAD

**Discussion by StrictlyMonad** [Slackadays/Clipboard#151] [Q&A]
Back here with a question that has been bothering me for quite some time. How can I make it so that a simple script which utilizies `maim` and `Clipboard` is able to copy the captured image to the system clipboard?

Currently `cb` is working as intended, but the only thing I am not able to do is get XMonad to copy the output of screenshot to system-clipboard for me to later paste around. I know that only the `cb` part is messed up (works fine when executed through terminal though), because the notification function I have defined in the script is returning the correct path to the image with the thumbnail set to it too! While `cb` returns `/run/user/1000/Clipboard/0/data/14/rawdata.clipboard` on paste and `cat rawdata.clipboard` returns empty.

**Comment by Slackadays** [comment]
What does the script look like? If I can reproduce it myself, then I can probably see what's happening exactly. Also, note that desktop script runners like GNOME appear to not allocate a tty, so CB thinks it's getting piped for all actions all the time.

**Reply by StrictlyMonad** [reply]
The script looks like:
```perl
#!/usr/bin/env perl

use strict;
use warnings;
use Getopt::Long;
use File::Basename;
use File::Copy;
use POSIX qw(strftime);
use Cwd 'abs_path';

my $help_message = "Failed to capture screen.\nPlease, try again!";

my $workspace = 0;
my $active_window = 0;
my $selection = 0;
my $command;

GetOptions(
    "workspace"     => \$workspace,
    "active-window" => \$active_window,
    "selection"     => \$selection,
) or die $help_message;

$command = shift @ARGV;

my ($title, $parameter);

if ($workspace) {
    $title = "Workspace";
    $parameter = "";
}
elsif ($active_window) {
    $title = "Active-Win";
    $parameter = "-i \$(xdotool getactivewindow)";
}
elsif ($selection) {
    $title = "Selection";
    $parameter = "-s";
}
else {
    die $help_message;
}

sub notify {
    my ($image, $title, $message) = @_;
    system(
        "notify-send",
        "-t", "750",
        "-h", "string:synchronous:screenshot",
        "-i", $image,
        "-u", "low",
        $title,
        $message,
    );
}

sub capture_screen {
    my $current_datetime = strftime("%Y-%m-%d_%H-%M-%S", localtime);
    my $filename = $title . "_" . $current_datetime . ".png";

    if ($command eq "system") {
        my $filepath = abs_path("~/Pictures/Screenshots/" . $filename);

        my $maim_cmd = "maim -u $parameter $filepath";
        system($maim_cmd);

        notify($filepath, "Archived Screenshot!", " $current_datetime");
    }
    elsif ($command eq "clipboard") {
        my $filepath = "/tmp/" . $filename;

        my $maim_cmd = "maim -u $parameter $filepath";
        system($maim_cmd);

        my $clipboard_cmd = "cb copy $filepath";
        system($clipboard_cmd);

        notify($filepath, "Clipped Screenshot!", "⎘ $current_datetime");
    }
    else {
        die $help_message;
    }
}

capture_screen();
```

**Reply by StrictlyMonad** [reply]
Example command: `./perlCapt clipboard --selection`

**Comment by StrictlyMonad** [comment]
@Slackadays were you able to replicate the issue that I was facing?

**Reply by Slackadays** [reply]
I've been super busy right now so I've had 0 time to even think about CB, but I will take a look at this as soon as I can! :)

**Reply by StrictlyMonad** [reply]
No worries! Just wanted to make sure if you got the notification or not. Appreciate the work mate and have a good day! :)

**Comment by StrictlyMonad** [comment]
Don't know what changed after updating my laptop, but somehow I am now able to copy/paste contents with the same script as I had pasted before. :man_shrugging: 

I honestly don't have time to bisect the change that caused this fix, but if I ever stumble upon it then I'll make sure to mention it here!
